data-prepper-api/src/main/java/org/opensearch/dataprepper/model/document/Document.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.model.document;
+
+import org.opensearch.dataprepper.model.event.Event;
+
+/**
+ * A document event in Data Prepper represents a document type.
+ * @since 2.1
+ */
+public interface Document extends Event {
```

Comment:
Why do we need a new Document event type? What feature's are we planning to add here? Are there special Document features we are planning to support and need standardized? 

I want to make sure we aren't building one off event types that will not be re-used. The S3 source plugin read documents from S3, yet we are using the generic event type. Ref: https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3ObjectWorker.java#L66

Reply:
Good question. The reason we want to add `Document` event type is to support full-text search use case. The converted RSS items that the user would want to use are documents which correspond to webpages from the RSS feed. Currently, the `EventType` enum supports only 3 use cases: Log, Trace, Metric. The RSS docs do not fit into any of these categories. By introducing a Document type, we can fit other future text use cases in this EventType, making it extensible. Details about which plugins to support are still under way but it did come up in a design discussion.

As for S3, while building the `Event` it does end up building a `JacksonLog`. There's 3 implementations of the parse method that ends up creating instances of `JacksonLog`:
* [JsonCodec](https://github.com/opensearch-project/data-prepper/blob/8cd0f870afe74dca34187a6d31034c166ec44d48/data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/codec/JsonCodec.java#L57)
* [CsvCodec](https://github.com/opensearch-project/data-prepper/blob/8cd0f870afe74dca34187a6d31034c166ec44d48/data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/codec/CsvCodec.java#L85)
* [NewlineDelimitedCodec](https://github.com/opensearch-project/data-prepper/blob/8cd0f870afe74dca34187a6d31034c166ec44d48/data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/codec/NewlineDelimitedCodec.java#L76)

Reply:
Thanks for the explanation. I have couple of questions Since the underlying object is still an Event(which is essentially a map), what would be stored in the map for the document? How does that help with "full text search"

Reply:
@kkondaka - Each RSS `item` will be a corresponding `Event` as mentioned in the Design doc. Since we need a new `EventType` to categorize RSS documents, I think it makes sense to have a corresponding Document interface to extend `Event` even though there's no required feature at the moment.